prefect2/data_source_work_queues.go

```diff
@@ -0,0 +1,140 @@
+package prefect2
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	hc "terraform-provider-prefect2/prefect2_api"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceWorkQueues() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceWorkQueuesRead,
+		Schema: map[string]*schema.Schema{
+			"workspace_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"work_queue_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"work_queues": &schema.Schema{
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"created": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"updated": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"name": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"description": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"is_paused": &schema.Schema{
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"concurrency_limit": &schema.Schema{
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"priority": &schema.Schema{
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"work_pool_id": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"filter": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"last_polled": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceWorkQueuesRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	c := m.(*hc.Client)
```

Comment:
I'm not too familiar with the Terraform plugin infrastructure, is it possible that they pass us a different parameter? Do we need to check this type assertion to prevent a panic, or can we expect that `m` is always a Client?

Reply:
I put a helper function now that is uses for all these methods. The helper function `getClient` checks for the type.